### PR TITLE
docs: Improved Model and ModelRc documentation

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: internal/core/model.rs [hunk 1]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
 /// The `ModelRc` struct holds something that implements the [`Model`] trait.
 /// This is used in `for` expressions in the .slint language.
 /// Array properties in the .slint language are holding a ModelRc.
+/// For example, a `property <[string]> foo` will be of type `ModelRc<SharedString>`
+/// and behind the scene, wraps a `Rc<dyn Model<Data = SharedString>>`
```
Comment (tronical) [inline internal/core/model.rs:512]:
```suggestion
/// and, behind the scenes, wraps a `Rc<dyn Model<Data = SharedString>>.`
```

File: internal/core/model.rs [hunk 2]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+///         callback add_item <=> btn.clicked;
+///         in property <[string]> the_model;
+///         HorizontalLayout {
+///             for it in the_model : Text { text: it; }
+///             btn := Button { text: "Add"; }
+///         }
+///     }
+/// }
+/// let ui = Example::new().unwrap();
+/// // create a VecModel and put it in a Rc
```
Comment (tronical) [inline internal/core/model.rs:537]:
```suggestion
/// // Create a VecModel and put it in an Rc.
```

File: internal/core/model.rs [hunk 3]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+/// let the_model : Rc<VecModel<SharedString>> =
+///         Rc::new(VecModel::from(vec!["Hello".into(), "World".into()]));
+/// // we can convert it to a ModelRc
```
Comment (tronical) [inline internal/core/model.rs:540]:
```suggestion
/// // Convert it to a ModelRc.
```

File: internal/core/model.rs [hunk 4]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+/// let the_model_rc = ModelRc::from(the_model.clone());
+/// // it can be set on our ui, the generated set_the_model setter from the
```
Comment (tronical) [inline internal/core/model.rs:542]:
```suggestion
/// // Pass the model to the ui: The generated set_the_model setter from the
```

File: internal/core/model.rs [hunk 5]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+/// // the_model property takes a ModelRc
```
Comment (tronical) [inline internal/core/model.rs:543]:
```suggestion
/// // the_model property takes a ModelRc.
```

File: internal/core/model.rs [hunk 6]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+/// ui.set_the_model(the_model_rc);
+///
+/// // we have kept a strong reference to the_model, we can modify it in a callback
```
Comment (tronical) [inline internal/core/model.rs:546]:
```suggestion
/// // We have kept a strong reference to the_model, to modify it in a callback.
```

File: internal/core/model.rs [hunk 7]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+/// ui.on_add_item(move || {
+///     // Uses VecModel API, VecModel uses the Model notification mechanism to let Slint
```
Comment (tronical) [inline internal/core/model.rs:548]:
```suggestion
///     // Use VecModel API: VecModel uses the Model notification mechanism to let Slint
```

File: internal/core/model.rs [hunk 8]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+///     // know it needs to refresh the UI
```
Comment (tronical) [inline internal/core/model.rs:549]:
```suggestion
///     // know it needs to refresh the UI.
```

File: internal/core/model.rs [hunk 9]
```diff
@@ -502,12 +508,58 @@ impl Model for bool {
...
+///     the_model.push("SomeValue".into());
+/// });
+///
+/// // Alternative: we can re-use a getter
```
Comment (tronical) [inline internal/core/model.rs:553]:
```suggestion
/// // Alternative: we can re-use a getter.
```

**Comment by colelawrence** [comment]
This definitely helps fill out the details for me. I like the additional note that you can get the model again with the getter `.get_the_model()`

**Pull request merged**